File: Problemset/1409-Queries on a Permutation With Key/Simulation.rb
# @param {Integer[]} queries
# @param {Integer} m
# @return {Integer[]}
def process_queries(queries, m)
    p = Array(m.downto(1))
    ret = Array.new(queries.length)

    for i in 0...queries.length
        ret[i] = m - 1 - p.index(queries[i])
        p.delete(queries[i])
        p.push(queries[i])
    end

    return ret
end

File: Problemset/1409-Queries on a Permutation With Key/Simulation.rs
impl Solution {
    pub fn process_queries(queries: Vec<i32>, m: i32) -> Vec<i32> {
        let mut p = (1..=m).rev().collect::<Vec<i32>>();
        let mut ret = vec![0; queries.len()];

        for i in 0..queries.len() {
            let posi = p.iter().position(|&x| x == queries[i]).unwrap();
            let x = p.remove(posi);
            p.push(x);
            ret[i] = m - 1 - posi as i32;
        }

        ret
    }
}
